File: LeetCodeCPP/1431.kids-with-the-greatest-number-of-candies.cpp
/*
 * @lc app=leetcode id=1431 lang=cpp
 *
 * [1431] Kids With the Greatest Number of Candies
 */

#include <vector>
#include <algorithm>

using namespace std;

// @lc code=start
class Solution
{
public:
    vector<bool> kidsWithCandies(vector<int> &candies, int extraCandies)
    {
        vector<bool> result;
        int max = *max_element(candies.begin(), candies.end());
        for (auto i : candies)
            result.push_back((extraCandies + i >= max));
        return result;
    }
};
// @lc code=end

File: LeetCodePython/1431.kids-with-the-greatest-number-of-candies.py
#
# @lc app=leetcode id=1431 lang=python3
#
# [1431] Kids With the Greatest Number of Candies
#
from typing import List


# @lc code=start
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        result: List[bool] = []
        max_candies = max(candies)
        for candy in candies:
            result.append(extraCandies + candy >= max_candies)
        return result


# @lc code=end
